lib/netsuite/records/gift_cert_redemption.rb

```diff
@@ -0,0 +1,24 @@
+module NetSuite
+  module Records
+    class GiftCertRedemption
+      include Support::Fields
+      include Support::RecordRefs
+      include Support::Records
+      include Support::Actions
+      include Namespaces::TranSales
+
+      actions :get, :get_list, :add, :initialize, :delete, :update, :upsert, :search
+
+      fields :auth_code_amt_remaining, :auth_code_applied, :gift_cert_available
+
+      record_refs :auth_code
+
+      attr_reader :internal_id
+
+      def initialize(attributes = {})
+        initialize_from_attributes_hash(attributes)
```

Comment:
@SEAnderson07 You need to pull the internal ID manually like this: https://github.com/NetSweet/netsuite/blob/master/lib/netsuite/records/credit_memo.rb#L36-L37